### ISSUE TITLE
Opposing Force - Low Definition MP5 animations mismatch

### ISSUE THREAD

**Issue by JoelTroch** [ValveSoftware/halflife#1620]
In the low definition model of Opposing Force's MP5, there are 2 deploy animations whereas only one is present in the HD model. The issue is that when we fire the LD MP5, the second deploy animation is played instead of the first shoot animation and the third shoot animation is never played.

I have fixed this issue by deleting the unused second deploy animation in the MDL file, you can download the fixed MDL here : http://www.mediafire.com/download/uzq24lp539pkxzi/v_9mmar.mdl

**Comment by malortie** [comment]
@JoelTroch

This might be late for a reply but still...

An alternate method for fixing this without recompiling models
would be to do the following:

In **ev_hldm.h**

Add the following enum definition under *mp5_e*.

```cpp
enum mp5hd_e
{
	MP5HD_LONGIDLE = 0,
	MP5HD_IDLE1,
	MP5HD_LAUNCH,
	MP5HD_RELOAD,
	MP5HD_DEPLOY,
	MP5HD_DEPLOY2,
	MP5HD_FIRE1,
	MP5HD_FIRE2,
	MP5HD_FIRE3,
};
```

In **ev_hldm.cpp**

In function void `EV_FireMP5( struct event_args_s *args  )`, edit
the following lines:

Before:

```cpp
if ( EV_IsLocal( idx ) )
{
	// Add muzzle flash to current weapon model
	EV_MuzzleFlash();
	gEngfuncs.pEventAPI->EV_WeaponAnimation( MP5_FIRE1 + gEngfuncs.pfnRandomLong(0,2), 2 );

	V_PunchAxis( 0, gEngfuncs.pfnRandomFloat( -2, 2 ) );
}
```

After:


```cpp
if ( EV_IsLocal( idx ) )
{
	// Add muzzle flash to current weapon model
	EV_MuzzleFlash();
	
	// Retrieve a pointer to hi definition cvar.
	cvar_t* pHiRes =  IEngineStudio.GetCvar( "cl_himodels" );
	
	// Play correct weapon animation depending if using LD/HD model.
	if ( pHiRes && pHiRes->value )
		gEngfuncs.pEventAPI->EV_WeaponAnimation( MP5HD_FIRE1 + gEngfuncs.pfnRandomLong(0,2), 2 );
	else
		gEngfuncs.pEventAPI->EV_WeaponAnimation( MP5_FIRE1 + gEngfuncs.pfnRandomLong(0,2), 2 );

	V_PunchAxis( 0, gEngfuncs.pfnRandomFloat( -2, 2 ) );
}
```

I currently don't have the time to test it, but I am quite sure it would work.
If you are still looking at this issue, and if I still haven't tested it, please
let me know if it works.

I hope this helps.


Also, it appears to be a problem on _Blue Shift_ as well.

Since _Gearbox_ reused the same animations for weapons (simple skin change), this problem still
persists on _Blue Shift_.

**Comment by JoelTroch** [comment]
Haven't tested and don't have time to do it, maybe the CVAR's name is `_himodels` ?

**Comment by tschumann** [comment]
I might be wrong but I thought cl_himodels was for multiplayer player models. I'm not sure if there is a way to detect whether the HD pack is in use (though I'd be interested to know).

**Comment by JoelTroch** [comment]
I'm a fucking idiot. I realized just now that I've already done a "HD/addons folders enabled" detector for ARRANGEMENT months ago.

On Windows, you have to read the registry. To be more precise, the keys `addons_folder` and `hdmodels` located in `HKEY_CURRENT_USER\SOFTWARE\Valve\Half-Life\Settings`.

On Linux (and maybe OS X), you don't have a registry like Windows. So you have to parse the `hl.conf` file located in the Half-Life folder. The file looks something like that :
```
addons_folder=0
himodels=1
```

**Comment by malortie** [comment]
@tschumann 

> I thought cl_himodels was for multiplayer player models

Correct. My apologies for this mistake. I have not worked as much as you on the multiplayer side of this engine.
